### PR TITLE
CAPA

### PR DESCRIPTION
**Pull request by whites11**
Towards: https://github.com/giantswarm/roadmap/issues/1546

This PR is the implementation of domain alias feature for CAPA.

During tests:
- the creation of irsa resources (cloudfront, acm, route53) seems fine
- the secret generated contained the expected data
- the existing clusters' API server pods didn't have the right irsa settings
- it was not possible to run a test cluster

Consequence: Testing is only partial

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Review by calvix — APPROVED** [review]
looks sane,  LGTM if properly tested

**Comment by whites11** [comment]
> looks sane, LGTM if properly tested

full testing is impossible right now to me. I wasn't able to create a cluster.
I have no idea how the last step between the secret containing irsa data and the API server flags is supposed to work (it does not at the moment) and thus I did only basic testing (ensuring irsa operator does what it's meant to.

File: pkg/irsa/capa/capa.go [hunk 1]
```diff
@@ -317,3 +449,50 @@ func (s *Service) ServiceAccountSecret(ctx context.Context) (*rsa.PrivateKey, er
...
+	baseDomain := cv.BaseDomain
+	if baseDomain == "" {
+		return "", microerror.Mask(baseDomainNotFound)
+	}
+
+	return baseDomain, nil
+}
+
+func (s *Service) getCloudFrontAliasDomain(ctx context.Context, baseDomain string) (string, error) {
+	awscluster := &v1alpha3.AWSCluster{}
```
Comment (njuettner) [inline pkg/irsa/capa/capa.go:485]: This must be `v1beta1`.

**Review by njuettner — CHANGES_REQUESTED** [review]

File: pkg/irsa/capa/capa.go (hunk 1 above)
Comment (whites11) [reply pkg/irsa/capa/capa.go]: bumped dependency and using v1beta1 now

**Review by njuettner — APPROVED** [review]
lgtm 🤗

**Comment by paurosello** [comment]
LGTM

**Pull request merged**